docs/src/tutorial/pure.rst

```diff
@@ -1,512 +1,687 @@
-Cython will generate C code that uses a C array of C ints.
+.. _pure-mode:
+
+Pure Python Mode
+================
+
+.. include::
+    ../two-syntax-variants-used
+
+In some cases, it's desirable to speed up Python code without losing the
+ability to run it with the Python interpreter.  While pure Python scripts
+can be compiled with Cython, it usually results only in a speed gain of
+about 20%-50%.
+
+To go beyond that, Cython provides language constructs to add static typing
+and cythonic functionalities to a Python module to make it run much faster
+when compiled, while still allowing it to be interpreted.
+This is accomplished via an augmenting ``.pxd`` file, via Python
+type :ref:`pep484_type_annotations` (following
+`PEP 484 <https://www.python.org/dev/peps/pep-0484/>`_ and
+`PEP 526 <https://www.python.org/dev/peps/pep-0526/>`_), and/or
+via special functions and decorators available after importing the magic
+``cython`` module.  All three ways can be combined at need, although
+projects would commonly decide on a specific way to keep the static type
+information easy to manage.
+
+Although it is not typically recommended over writing straight Cython code
+in a :file:`.pyx` file, there are legitimate reasons to do this - easier
+testing and debugging, collaboration with pure Python developers, etc.
+In pure mode, you are more or less restricted to code that can be expressed
+(or at least emulated) in Python, plus static type declarations. Anything
+beyond that can only be done in .pyx files with extended language syntax,
+because it depends on features of the Cython compiler.
+
+.. _augmenting_pxd:
+
+Augmenting .pxd
+---------------
+
+Using an augmenting :file:`.pxd` allows to leave the original :file:`.py` file completely untouched. On the other hand, one needs to maintain both the :file:`.pxd` and the :file:`.py` to keep them in sync.
+
+While declarations in a :file:`.pyx` file must correspond exactly with those of a :file:`.pxd` file with the same name (and any contradiction results in a compile time error, see :doc:`pxd_files`), the untyped definitions in a :file:`.py` file can be overridden and augmented with static types by the more specific ones present in a :file:`.pxd`.
+
+If a :file:`.pxd` file is found with the same name as the :file:`.py` file being compiled, the compiler will convert the corresponding definitions in the :file:`.py` file to be of the declared type. The following tabs illustrate the process, showing the original Python file and the final equivalent Cython code, which are both defined by the declarations in the supplementing :file:`.pxd` file:
+
+.. tabs::
+
+   .. group-tab:: Pure Python
+
+      .. literalinclude:: ../../examples/tutorial/pure/A.py
+         :caption: A.py
+
+   .. group-tab:: Cython
+
+      .. literalinclude:: ../../examples/tutorial/pure/A_equivalent.pyx
+         :caption: A_equivalent.pyx
+
+The required declarations are defined in the supplementing :file:`A.pxd` file:

```

Comment:
I think this is now a bit confusing - the A.pxd file is only required for the pure Python version.